src/content/docs/recipes/apps/spawn-vim.md

```diff
@@ -0,0 +1,113 @@
+`Action` based on the key pressed. Here, we are calling `run_editor` function on `Action::EditFile`
+which we will define in next section.
+
+## Spawning Vim
+
+Now, let's define the function `run_editor` function attached to `Action::EditFile` action.
+
+```rust title="main.rs"
+{{ #include @code/recipes/how-to-spawn-vim/src/main.rs:run_editor }}
+```
+
+To spawn Vim from our TUI app, we first need to relinquish control of input and output, allowing Vim
+to have full control over the terminal.
+
+The `run_editor` function handles the logic for spawning vim. First, we leave the alternate screen
+and disable raw mode to restore terminal to it's original state. This part is similar to what
+[`ratatui::restore`](https://docs.rs/ratatui/latest/ratatui/fn.restore.html) function does in
+the `main` function. Next, we spawn a child process with
+`Command::new("vim").arg("/tmp/a.txt").status()` which launches `vim` to edit the given file. At
+this point, we have given up control of our TUI app to vim. Our TUI app will now wait for the exit
+status of the child process. Once the user exits Vim, our TUI app regains control over the terminal
+by re-entering alternate screen and enabling raw mode. Lastly, we clear the terminal to ensure the
+TUI is displayed correctly.
+
+:::note
+
+Before running another application from your app, you must relinquish control of input and output,
+allowing the other app to function correctly.
+
+In the example above, we use a simple event-handling setup. However, if you are using advanced
+setups like [component template](https://github.com/ratatui-org/templates), you will need to pause
+input events before spawning an external process like Vim. Otherwise, Vim won't have full control
+over keybindings and it won't work properly.
+
+Using the
+[`tui` module](https://github.com/ratatui-org/templates/blob/5e823efc871107345d59e5deff9284235c1f0bbc/component/template/src/tui.rs)
+of the component template, you can do something like this to pause and resume event handlers:
+
+```rust
+Action::EditFile => {
+  tui.exit()?;
+  let cmd = String::from("vim");
+  let cmd_args = vec!["/tmp/a.txt".into()];
+  let status = std::process::Command::new(&command).args(&args).status()?;
+  if !status.success() {
+    eprintln!("\nCommand failed with status: {}", status);
+  }
+  tui.enter()?;
+  tui.terminal.clear();
+}
+```
+
+One more thing to note is that when attempting to start an external process without using the
+pattern in the component template, issues can arise such as ANSI RGB values being printed into the
+TUI on returning back from external process. If you encounter such issues, please refer to
+[orhun/rattler-build@84ea16a](https://github.com/orhun/rattler-build/commit/84ea16a4f5af33e2703b6330fcb977065263cef6)
+and [kdheepak/taskwarrior-tui#46](https://github.com/kdheepak/taskwarrior-tui/issues/46). Using
+`select!` + `cancellation_token` + `tokio` as in the component template avoids this problem.
```

Comment:
Might be useful to mention the reason the RGB values are being printed to the terminal is because `vim` requests the terminal background color, and when the terminal responds over `stdin`, those are being read by crossterm instead.

Reply:
I have added this change along with mentioning [`editor-command`](https://docs.rs/editor-command/latest/editor_command) in tips section from this issue comment https://github.com/ratatui/ratatui-website/issues/406#issuecomment-2325354730